### PR TITLE
Swipe terraform integration.

### PR DESCRIPTION
**Pull request by jgadling**
### Description

Write a description of your pull request here

#### Issue
[ch<fill_in_issue_number>](https://app.clubhouse.io/genepi/story/<fill_in_issue_number>)

### Test plan

Write how your changes are tested, or give a convincing reason why they can't be tested automatically.

### CODE REVIEW

File: .happy/terraform/modules/batch/main.tf [hunk 1]
```diff
@@ -3,20 +3,26 @@
...
-  container_properties = jsonencode(yamldecode(templatefile("${path.module}/container_properties.yml", {
+locals {
+  container_properties = yamldecode(templatefile("${path.module}/container_properties.yml", {
     app_name           = var.app_name,
-    batch_docker_image = var.image
+    batch_docker_image = var.swipe_image
     aws_region         = data.aws_region.current.name,
     batch_job_role_arn = var.batch_role_arn
     deployment_stage   = var.deployment_stage
+    deployment_environment = "aspen-rdev"
```
Comment (jgadling) [inline .happy/terraform/modules/batch/main.tf]: @jgadling fix this

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 2]
```diff
@@ -27,9 +27,23 @@ locals {
...
+  aspen_data_bucket     = local.secret["s3_buckets"]["aspen_data"]["name"]
+
+  # Web images
   frontend_image_repo   = local.secret["ecrs"]["frontend"]["url"]
   backend_image_repo    = local.secret["ecrs"]["backend"]["url"]
-  nextstrain_image_repo = local.secret["ecrs"]["nextstrain"]["url"]
+
+  # Workflow images
+  pangolin_image_repo   = local.secret["ecrs"]["pangolin"]["url"]
+  phylotree_image_repo  = local.secret["ecrs"]["phylotree"]["url"]
```
Comment (phoenixAja) [inline .happy/terraform/modules/ecs-stack/main.tf]: nextstrain instead of phylotree?

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 3]
```diff
@@ -162,12 +194,13 @@ module delete_db {
...
-module nextstrain_batch {
+module swipe_batch {
   source                = "../batch"
-  app_name              = "nextstrain"
+  app_name              = "swipe"
   stack_resource_prefix = local.stack_resource_prefix
-  image                 = "${local.nextstrain_image_repo}:${local.image_tag}"
+  image                 = "${local.swipe_image_repo}:latest" # TODO hardcoded
   batch_role_arn        = local.batch_role_arn
+  swipe_image            = "${local.swipe_image_repo}:rev-3" # FIXME rev shouldn't be hardcoded
```
Comment (phoenixAja) [inline .happy/terraform/modules/ecs-stack/main.tf]: don't need comments anymore?

File: .happy/terraform/modules/sfn_config/gisaid.wdl [hunk 4]
```diff
@@ -0,0 +1,275 @@
+version 1.1
+
+workflow LoadGISAID {
+    input {
+        String docker_image_id = "aspen-gisaid"
+        String aws_region = "us-west-2"
+        String db_data_bucket = "aspen-data-rdev"
+        String gisaid_ndjson_export_url = "https://www.epicov.org/epi3/3p/exp3/export/export.json.bz2"
+        String gisaid_ndjson_staging_bucket = "aspen-data-rdev"
```
Comment (phoenixAja) [inline .happy/terraform/modules/sfn_config/gisaid.wdl:9]: change this to a different default?

File: .happy/terraform/modules/sfn_config/main.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
+resource "aws_s3_bucket_object" "wdl" {
+  bucket = var.swipe_wdl_bucket
+  key    = "${var.remote_dev_prefix}/${basename(var.wdl_path)}-v0.0.1.wdl" # Swipe lambdas require specially formatted filenames.
```
Comment (phoenixAja) [inline .happy/terraform/modules/sfn_config/main.tf:3]: wierd

File: .happy/terraform/modules/swipe-sfn/main.tf [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  sfn_def = yamldecode(templatefile("${path.module}/sfn.yml", merge(local.lambdas, {
+    deployment_environment = local.env
+    deployment_stage = var.deployment_stage
+    remote_dev_prefix = var.remote_dev_prefix
+    aws_default_region = "us-west-2" # FIXME hardcoded
+    batch_ec2_job_queue_name = var.ec2_queue_arn
+    batch_spot_job_queue_name = var.spot_queue_arn
+    batch_job_definition_name = var.job_definition_name
+    memory = 420000
+    vcpus = 1
```
Comment (phoenixAja) [inline .happy/terraform/modules/swipe-sfn/main.tf:25]: this seems low?

File: Makefile [hunk 7]
```diff
@@ -116,6 +116,11 @@ local-rebuild: .env.ecr local-ecr-login ## Rebuild local dev without re-importin
 	docker-compose $(COMPOSE_OPTS) build frontend backend utility
 	docker-compose $(COMPOSE_OPTS) up -d
 
+.PHONY: local-rebuild-workflows
+local-rebuild-workflows: .env.ecr local-ecr-login ## Rebuild batch containers
+	docker-compose $(COMPOSE_OPTS) build gisaid pangolin
```
Comment (phoenixAja) [inline Makefile]: add nextstrain?

**Review by phoenixAja — APPROVED** [review]

**Pull request merged**